pkg/model/model.go

```diff
@@ -17,72 +17,227 @@ limitations under the License.
+// The default power model type is Ratio
+func getPowerModelType(powerSourceTarget string) (modelType types.ModelType) {
+	useEstimatorSidecarStr := config.ModelConfigValues[getModelConfigKey(powerSourceTarget, config.EstimatorEnabledKey)]
+	if strings.EqualFold(useEstimatorSidecarStr, "true") {
+		modelType = types.EstimatorSidecar
+		return
+	}
+	useLinearRegressionStr := config.ModelConfigValues[getModelConfigKey(powerSourceTarget, config.LinearRegressionEnabledKey)]
+	if strings.EqualFold(useLinearRegressionStr, "true") {
+		modelType = types.LinearRegressor
+		return
 	}
-	return valid, estimateFunc
+	// set the default node power model as LinearRegressor
+	if powerSourceTarget == config.NodePlatformPowerKey || powerSourceTarget == config.NodeComponentsPowerKey {
+		modelType = types.LinearRegressor
+		return
+	}
+	// set the default container power model as Ratio
+	modelType = types.Ratio
+	return
+}
+
+// getPowerModelName return the model name for a given power source, such as platform or components power sources
+func getPowerModelName(powerSourceTarget string) (modelName string) {
+	modelName = config.ModelConfigValues[getModelConfigKey(powerSourceTarget, config.FixedModelNameKey)]
+	return
+}
+
+// getPowerModelFilter return the model filter for a given power source, such as platform or components power sources
+// The model filter is used to select a model, for example selecting a model with the acceptable error: 'mae:0.5'
+func getPowerModelFilter(powerSourceTarget string) (selectFilter string) {
+	selectFilter = config.ModelConfigValues[getModelConfigKey(powerSourceTarget, config.ModelFiltersKey)]
+	return
+}
+
+// getPowerModelDownloadURL return the url to download the pre-trained power model for a given power source, such as platform or components power sources
+func getPowerModelDownloadURL(powerSourceTarget string) (url string) {
+	url = config.ModelConfigValues[getModelConfigKey(powerSourceTarget, config.InitModelURLKey)]
+	return
 }
 
-func InitModelConfig(modelItem string) types.ModelConfig {
-	useEstimatorSidecar, selectedModel, selectFilter, initModelURL := config.GetModelConfig(modelItem)
-	modelConfig := types.ModelConfig{UseEstimatorSidecar: useEstimatorSidecar, SelectedModel: selectedModel, SelectFilter: selectFilter, InitModelURL: initModelURL}
-	klog.V(3).Infof("Model Config %s: %+v", modelItem, modelConfig)
-	return modelConfig
+// getPowerModelOutputType return the model output type for a given power source, such as platform, components, container or node power sources.
+// getPowerModelOutputType only affects LinearRegressor or EstimatorSidecar model. The Ratio model does not download data from the Model Server.
+func getPowerModelOutputType(powerSourceTarget string, modelType types.ModelType) types.ModelOutputType {
+	// Linear regression power model downloads the model weights
+	if modelType == types.LinearRegressor {
```

Comment:
Need to update to new model type (AbsPower and DynPower).

Reply:
Can you fix this in another PR?